src/searchengine/nova3/helpers.py

```diff
@@ -89,10 +89,10 @@ def entity2char(m: re.Match[str]) -> str:
     return re.sub(r'&#x(\w+);', lambda x: chr(int(x.group(1), 16)), t)
 
 
-def retrieve_url(url: str, custom_headers: Mapping[str, Any] = {}) -> str:
+def retrieve_url(url: str, custom_headers: Mapping[str, Any] = {}, post_data: Optional[bytes] = None) -> str:
```

Comment:
I would suggest `Optional[Any]` to shift the responsibility to the plugin that are using it, unless that exact type could be easily specified.
```suggestion
def retrieve_url(url: str, custom_headers: Mapping[str, Any] = {}, post_data: Optional[Any] = None) -> str:
```

Reply:
Urllib only wants bytes or bytes-like. Specifying Any here isn't terribly helpful to the plugin author, IMHO, who will be left having to find the type by trial and error, no?

Reply:
>Urllib only wants bytes or bytes-like. Specifying Any here isn't terribly helpful to the plugin author,

Seems a lot more than that to me:
https://github.com/python/typeshed/blob/e2aea1e70f60fcbb77f944a3dfd2a58dfb54200e/stdlib/urllib/request.pyi#L53

>IMHO, who will be left having to find the type by trial and error, no?

IMO the plugin author is likely to dig up the source code and see why it is `Any` and find out how it is used.

I really don't want to maintain the types when later the upstream added something new. Or later another another plugin author wants to add support for some type other than `bytes | None`.

I rather it being `Any` than providing insufficient/incorrect typing.